tool_helper_utils/__init__.py

```diff
@@ -2,12 +2,30 @@
 import sys
 
 
+class AbstractBaseFactoryFactory:
```

Comment:
This should really derive from `abc.ABC`, or at least set `metaclass=abc.ABCMeta`.